dbt-snowflake/src/dbt/adapters/snowflake/impl.py

```diff
@@ -268,22 +268,39 @@ def list_relations_without_caching(
 
         # this can be collapsed once Snowflake adds is_iceberg to show objects
         columns = ["database_name", "schema_name", "name", "kind", "is_dynamic"]
-        if self.behavior.enable_iceberg_materializations.no_warn:
+
+        # If the is_iceberg column is already on schema_objects query, we can ignore
+        # normalizing using our homemade is_iceberg
+        if (
```

Comment:
I don't think this conditional is quite right. We only want to do any renaming if iceberg materializations are enabled. And then within that condition, we either need to rename the all caps version, as is currently the process, or rename the dbt-calculated columns, as you do below. And then we only append `is_iceberg` if iceberg materializations are enabled. The if block below will append it when it exists, regardless of whether the user has enabled this flag.